app.js

```diff
@@ -1,208 +1,223 @@
 'use strict';
 
-let _ = require('lodash');
-let assert = require('assert');
-let logfmt = require('logfmt');
-let express = require('express');
-let through = require('through');
-let urlUtil = require('url');
-let StatsD = require('node-statsd');
-let basicAuth = require('basic-auth');
-let statsd = new StatsD(parseStatsdUrl(process.env.STATSD_URL));
-let app = module.exports = express();
-let allowedApps = loadAllowedAppsFromEnv();
-
-if (process.env.DEBUG) {
-  console.log('Allowed apps', allowedApps);
-  statsd.send = wrap(statsd.send.bind(statsd), console.log.bind(null, 'Intercepted: statsd.send(%s):'));
-}
+let throng = require('throng');
+let WORKERS = process.env.WEB_CONCURRENCY || 1;
+let port = process.env.PORT || 3000;
 
-app.use(logfmt.bodyParserStream());
-app.use(function authenticate (req, res, next) {
-  let auth = basicAuth(req) || {};
-  let app = allowedApps[auth.name];
-  if (app !== undefined && app.password === auth.pass) {
-    req.defaultTags = app.tags;
-    req.prefix = app.prefix;
-    next();
-  } else {
-    res.status(401).send('Unauthorized');
-    if (process.env.DEBUG) {
-      console.log('Unauthorized access by %s', auth.name);
-    }
-  }
-});
 
-app.post('/', function (req, res) {
-  if(req.body !== undefined) {
-    req.body.pipe(through(line => processLine(line, req.prefix, req.defaultTags)));
+if (require.main === module) {
```

Comment:
Try run npm test without this.
On Tue 11 Aug 2015 at 16:58 Jakub Korál <notifications@github.com> wrote:

> In app.js
> <https://github.com/apiaryio/heroku-datadog-drain/pull/1#discussion_r36754707>
> :
>
> >
> > -app.post('/', function (req, res) {
> > -  if(req.body !== undefined) {
> > -    req.body.pipe(through(line => processLine(line, req.prefix, req.defaultTags)));
> > +if (require.main === module) {
>
> what is this here for @abtris <https://github.com/abtris> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apiaryio/heroku-datadog-drain/pull/1/files#r36754707>.
>